crates/shared/src/order_quoting.rs

```diff
@@ -434,6 +454,59 @@ impl OrderQuoter {
 
         Ok(quote)
     }
+
+    /// Makes sure a quote was verified according to the configured rule.
+    async fn verify_quote(
+        &self,
+        estimate: &Estimate,
+        parameters: &QuoteParameters,
+        sell_amount: U256,
+    ) -> Result<(), CalculateQuoteError> {
+        if matches!(self.quote_verification, QuoteVerificationMode::Unverified) {
+            return Ok(()); // verification is not expected
+        }
+        let Some(verification) = &parameters.verification else {
+            return Ok(()); // no verification requested
+        };
```

Comment:
Is it possible to combine those two into one? And not have a verfication instruction on the QuoteParameters (which themselves depend on a config in the binary) and another config variable?

Reply:
IMO this problem originates from the fact that there are scenarios where we know verification will fail. To handle this I think we can only:
1. make `Verification` struct optional (or wrap it in an enum to give some additional context if needed)
2. always pass `Verification` struct and use default values when the user is not connected.

I didn't want to do `2` to not waste time simulating quotes we know will fail but maybe it ends up being cleaner over all to simply skip simulation if `from == 0x000` in the `TradeVerifier` than having to keep different implementation details in mind when reading code related to simulation stuff.

Reply:
@fleupold [aff63a1](https://github.com/cowprotocol/services/pull/2621/commits/aff63a1d0a2d5c5c090253cbe02584ecfb947684) is probably as close as we going to get with unifying verification handling for now.